gherkin/php/Makefile

```diff
@@ -0,0 +1,68 @@
+include default.mk
+
+GOOD_FEATURE_FILES = $(shell find testdata/good -name "*.feature")
+BAD_FEATURE_FILES  = $(shell find testdata/bad -name "*.feature")
+
+TOKENS       = $(patsubst testdata/%.feature,acceptance/testdata/%.feature.tokens,$(GOOD_FEATURE_FILES))
+ASTS         = $(patsubst testdata/%.feature,acceptance/testdata/%.feature.ast.ndjson,$(GOOD_FEATURE_FILES))
+PICKLES      = $(patsubst testdata/%.feature,acceptance/testdata/%.feature.pickles.ndjson,$(GOOD_FEATURE_FILES))
+SOURCES      = $(patsubst testdata/%.feature,acceptance/testdata/%.feature.source.ndjson,$(GOOD_FEATURE_FILES))
+ERRORS       = $(patsubst testdata/%.feature,acceptance/testdata/%.feature.errors.ndjson,$(BAD_FEATURE_FILES))
+
+.DELETE_ON_ERROR:
+
+default: .compared
+
+.compared: .deps $(TOKENS) $(SOURCES) $(ASTS) $(PICKLES) $(ERRORS)
+	touch $@
+
+acceptance/testdata/%.feature.tokens: testdata/%.feature testdata/%.feature.tokens
+	mkdir -p $(@D)
+	bin/gherkin-generate-tokens $< > $@
+	diff --unified $<.tokens $@
+
+acceptance/testdata/%.feature.source.ndjson: testdata/%.feature testdata/%.feature.source.ndjson .deps
+	mkdir -p $(@D)
+	bin/gherkin --predictable-ids --no-ast --no-pickles $< | jq --sort-keys --compact-output "." > $@
+	diff --unified <(jq "." $<.source.ndjson) <(jq "." $@)
+
+acceptance/testdata/%.feature.ast.ndjson: testdata/%.feature testdata/%.feature.ast.ndjson .deps
+	mkdir -p $(@D)
+	bin/gherkin --predictable-ids --no-source --no-pickles $< | jq --sort-keys --compact-output "." > $@
+	diff --unified <(jq "." $<.ast.ndjson) <(jq "." $@)
+
+acceptance/testdata/%.feature.pickles.ndjson: testdata/%.feature testdata/%.feature.pickles.ndjson .deps
+	mkdir -p $(@D)
+	bin/gherkin --predictable-ids --no-source --no-ast $< | jq --sort-keys --compact-output "." > $@
+	diff --unified <(jq "." $<.pickles.ndjson) <(jq "." $@)
+
+acceptance/testdata/%.feature.errors.ndjson: testdata/%.feature testdata/%.feature.errors.ndjson .deps
+	mkdir -p $(@D)
+	bin/gherkin --predictable-ids --no-source $< | jq --sort-keys --compact-output "." > $@
+	diff --unified <(jq "." $<.errors.ndjson) <(jq "." $@)
+
+
+.deps: build/classes.php
+
+build/classes.php: gherkin-php.razor gherkin.berp
+	$(berp-generate-parser)
+	csplit --quiet --prefix=build/Generated --suffix-format=%02d.php.tmp --elide-empty-files build/classes.php /^.*.php$$/ {*}
+	for file in build/Generated**; do mkdir -p src-generated/$$(head -n 1 $$file | sed 's/[^/]*.php$$//'); done
+	for file in build/Generated**; do tail -n +2 $$file > src-generated/$$(head -n 1 $$file); rm $$file; done
+	vendor/bin/php-cs-fixer --diff fix src-generated
+
+clean-build:
+	rm -rf build/classes.php
+	rm -rf src-generated/*
+
+clean:
+	rm -rf .compared acceptance
```

Comment:
I'm not sure how to make this also call the parent `clean`?

Reply:
I think it will call the "parent" clean target if you just add dependencies without altering the target, like with messages: https://github.com/cucumber/common/blob/main/messages/php/Makefile#L5